### DISCUSSION TITLE
Logic Apps Standard and High Availability using Availability Zones

### DISCUSSION THREAD

**Discussion by mark-abrams** [Azure/logicapps#887] [Q&A]
Logic Apps Standard support for High Availability using Availability Zones is described [here](https://learn.microsoft.com/en-us/azure/logic-apps/set-up-zone-redundancy-availability-zones?tabs=standard), this is the relevant information:

> Availability zone support is available for Standard logic apps, which are powered by Azure Functions extensibility. For more information, see [What is reliability in Azure Functions?](https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/reliability/reliability-functions.md#availability-zone-support).

> - You can enable availability zone redundancy only when you create Standard logic apps, either in a [supported Azure region](https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-functions/azure-functions-az-redundancy.md#requirements) or in an [App Service Environment v3 (ASE v3) - Windows plans only](https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/app-service/environment/overview-zone-redundancy.md). Currently, this capability supports only built-in connector operations, not Azure (managed) connector operations.

> - You can enable availability zone redundancy only for new Standard logic apps with workflows that run in single-tenant Azure Logic Apps. You can't enable availability zone redundancy for existing Standard logic app workflows.

> - You can enable availability zone redundancy only at creation time using Azure portal. No programmatic tool support, such as Azure PowerShell or Azure CLI, currently exists to enable availability zone redundancy.

In summary, it is not possible to create a highly available Standard Logic App using any Infrastructure as Code tooling (e.g. Bicep, Terraform) because the availability zone feature can only be enabled at creation time, using the Azure Portal. This is far from ideal, since most organisations use IaC tooling to deploy all Azure resources.

Can Microsoft confirm if this restriction is still in place, and if it is, are there any plans to address this so that Availability Zone support can be configured for **existing** Logic Apps, and configured via IaC tooling?

Mark

**Comment by mark-abrams** [comment]
Is anyone from Microsoft able to look and this and advise please? @divyaswarnkar , can you advise?

**Comment by jfastrom** [comment]
I'd say that zone availability can't really be considered a component feature until it's supported in automated deployment...
